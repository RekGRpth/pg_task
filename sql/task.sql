\unset ECHO
\set QUIET 1
-- Turn off echo and keep things quiet.

-- Format the output for nice TAP.
\pset format unaligned
\pset tuples_only true
\pset pager off

-- Revert all changes on failure.
\set ON_ERROR_ROLLBACK 1
\set ON_ERROR_STOP true

-- Load the TAP functions.
BEGIN;
--\i pgtap.sql
CREATE EXTENSION pgtap;

SELECT plan(94);

SELECT has_schema(current_setting('pg_task.default_schema', false)::name);

SELECT has_enum(current_setting('pg_task.default_schema', false)::name, 'state'::name);

SELECT enum_has_labels(current_setting('pg_task.default_schema', false)::name, 'state'::name, ARRAY['PLAN', 'TAKE', 'WORK', 'DONE', 'FAIL', 'STOP']);

SELECT has_table(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name);

SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'id'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('id') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'parent'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('parent') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'plan'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('plan') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'start'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('start') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'stop'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('stop') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'live'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('live') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'timeout'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('timeout') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'repeat'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('repeat') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'hash'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('hash') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'count'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('count') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'max'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('max') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'pid'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('pid') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'state'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('state') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'delete'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('delete') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'drift'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('drift') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'header'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('header') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'string'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('string') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'delimiter'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('delimiter') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'escape'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('escape') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'quote'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('quote') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'error'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('error') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'group'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('group') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'input'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('input') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'null'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('null') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'output'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('output') || ' should exist');
SELECT has_column(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'remote'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('remote') || ' should exist');

SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'id'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'plan'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'live'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'timeout'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'repeat'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'hash'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'count'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'max'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'state'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'delete'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'drift'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'header'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'string'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'delimiter'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'group'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'input'::name);
SELECT col_not_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'null'::name);

SELECT col_is_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'parent'::name);
SELECT col_is_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'start'::name);
SELECT col_is_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'stop'::name);
SELECT col_is_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'pid'::name);
SELECT col_is_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'escape'::name);
SELECT col_is_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'quote'::name);
SELECT col_is_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'error'::name);
SELECT col_is_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'output'::name);
SELECT col_is_null(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'remote'::name);

SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'id'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('id') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'parent'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('parent') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'plan'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('plan') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'live'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('live') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'timeout'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('timeout') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'repeat'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('repeat') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'hash'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('hash') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'count'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('count') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'max'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('max') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'state'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('state') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'delete'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('delete') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'drift'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('drift') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'header'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('header') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'string'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('string') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'delimiter'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('delimiter') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'group'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('group') || ' should have a default');
SELECT col_has_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'null'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('null') || ' should have a default');

SELECT col_hasnt_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'start'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('start') || ' should not have a default');
SELECT col_hasnt_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'stop'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('stop') || ' should not have a default');
SELECT col_hasnt_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'pid'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('pid') || ' should not have a default');
SELECT col_hasnt_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'escape'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('escape') || ' should not have a default');
SELECT col_hasnt_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'quote'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('quote') || ' should not have a default');
SELECT col_hasnt_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'error'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('error') || ' should not have a default');
SELECT col_hasnt_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'input'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('input') || ' should not have a default');
SELECT col_hasnt_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'output'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('output') || ' should not have a default');
SELECT col_hasnt_default(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'remote'::name, 'Column ' || quote_ident(current_setting('pg_task.default_table', false)) || '.' || quote_ident('remote') || ' should not have a default');

SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'id'::name, 'bigint'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'parent'::name, 'bigint'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'plan'::name, 'timestamp with time zone'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'start'::name, 'timestamp with time zone'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'stop'::name, 'timestamp with time zone'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'live'::name, 'interval'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'timeout'::name, 'interval'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'repeat'::name, 'interval'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'hash'::name, 'integer'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'count'::name, 'integer'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'max'::name, 'integer'::name);
SELECT col_type_is(current_setting('pg_task.default_schema', false)::name, current_setting('pg_task.default_table', false)::name, 'pid'::name, 'integer'::name);

SELECT * FROM finish();

ROLLBACK;
