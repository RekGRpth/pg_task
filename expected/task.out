\unset ECHO
1..153
ok 1 - Schema public should exist
ok 2 - Enum public.state should exist
ok 3 - Enum public.state should have labels (PLAN, TAKE, WORK, DONE, FAIL, STOP)
ok 4 - Table public.task should exist
ok 5 - Column task.id should exist
ok 6 - Column task.parent should exist
ok 7 - Column task.plan should exist
ok 8 - Column task.start should exist
ok 9 - Column task.stop should exist
ok 10 - Column task.live should exist
ok 11 - Column task.timeout should exist
ok 12 - Column task.repeat should exist
ok 13 - Column task.hash should exist
ok 14 - Column task.count should exist
ok 15 - Column task.max should exist
ok 16 - Column task.pid should exist
ok 17 - Column task.state should exist
ok 18 - Column task.delete should exist
ok 19 - Column task.drift should exist
ok 20 - Column task.header should exist
ok 21 - Column task.string should exist
ok 22 - Column task.delimiter should exist
ok 23 - Column task.escape should exist
ok 24 - Column task.quote should exist
ok 25 - Column task.error should exist
ok 26 - Column task."group" should exist
ok 27 - Column task.input should exist
ok 28 - Column task."null" should exist
ok 29 - Column task.output should exist
ok 30 - Column task.remote should exist
ok 31 - Column public.task.id should be NOT NULL
ok 32 - Column public.task.plan should be NOT NULL
ok 33 - Column public.task.live should be NOT NULL
ok 34 - Column public.task.timeout should be NOT NULL
ok 35 - Column public.task.repeat should be NOT NULL
ok 36 - Column public.task.hash should be NOT NULL
ok 37 - Column public.task.count should be NOT NULL
ok 38 - Column public.task.max should be NOT NULL
ok 39 - Column public.task.state should be NOT NULL
ok 40 - Column public.task.delete should be NOT NULL
ok 41 - Column public.task.drift should be NOT NULL
ok 42 - Column public.task.header should be NOT NULL
ok 43 - Column public.task.string should be NOT NULL
ok 44 - Column public.task.delimiter should be NOT NULL
ok 45 - Column public.task."group" should be NOT NULL
ok 46 - Column public.task.input should be NOT NULL
ok 47 - Column public.task."null" should be NOT NULL
ok 48 - Column public.task.parent should allow NULL
ok 49 - Column public.task.start should allow NULL
ok 50 - Column public.task.stop should allow NULL
ok 51 - Column public.task.pid should allow NULL
ok 52 - Column public.task.escape should allow NULL
ok 53 - Column public.task.quote should allow NULL
ok 54 - Column public.task.error should allow NULL
ok 55 - Column public.task.output should allow NULL
ok 56 - Column public.task.remote should allow NULL
ok 57 - Column task.id should have a default
ok 58 - Column task.parent should have a default
ok 59 - Column task.plan should have a default
ok 60 - Column task.live should have a default
ok 61 - Column task.timeout should have a default
ok 62 - Column task.repeat should have a default
ok 63 - Column task.hash should have a default
ok 64 - Column task.count should have a default
ok 65 - Column task.max should have a default
ok 66 - Column task.state should have a default
ok 67 - Column task.delete should have a default
ok 68 - Column task.drift should have a default
ok 69 - Column task.header should have a default
ok 70 - Column task.string should have a default
ok 71 - Column task.delimiter should have a default
ok 72 - Column task."group" should have a default
ok 73 - Column task."null" should have a default
ok 74 - Column task.start should not have a default
ok 75 - Column task.stop should not have a default
ok 76 - Column task.pid should not have a default
ok 77 - Column task.escape should not have a default
ok 78 - Column task.quote should not have a default
ok 79 - Column task.error should not have a default
ok 80 - Column task.input should not have a default
ok 81 - Column task.output should not have a default
ok 82 - Column task.remote should not have a default
ok 83 - Column public.task.id should be type bigint
ok 84 - Column public.task.parent should be type bigint
ok 85 - Column public.task.plan should be type timestamp with time zone
ok 86 - Column public.task.start should be type timestamp with time zone
ok 87 - Column public.task.stop should be type timestamp with time zone
ok 88 - Column public.task.live should be type interval
ok 89 - Column public.task.timeout should be type interval
ok 90 - Column public.task.repeat should be type interval
ok 91 - Column public.task.hash should be type integer
ok 92 - Column public.task.count should be type integer
ok 93 - Column public.task.max should be type integer
ok 94 - Column public.task.pid should be type integer
ok 95 - Column public.task.state should be type public.state
ok 96 - Column public.task.delete should be type boolean
ok 97 - Column public.task.drift should be type boolean
ok 98 - Column public.task.header should be type boolean
ok 99 - Column public.task.string should be type boolean
ok 100 - Column public.task.delimiter should be type char
ok 101 - Column public.task.escape should be type char
ok 102 - Column public.task.quote should be type char
ok 103 - Column public.task.error should be type text
ok 104 - Column public.task."group" should be type text
ok 105 - Column public.task.input should be type text
ok 106 - Column public.task."null" should be type text
ok 107 - Column public.task.output should be type text
ok 108 - Column public.task.remote should be type text
ok 109 - Column task.id should default to 'nextval(''task_id_seq''::regclass)'
ok 110 - Column task.parent should default to '(current_setting(''pg_task.id''::text, true))::bigint'
ok 111 - Column task.plan should default to 'CURRENT_TIMESTAMP'
ok 112 - Column task.live should default to '0 sec'
ok 113 - Column task.timeout should default to '0 sec'
ok 114 - Column task.repeat should default to '0 sec'
ok 115 - Column task.hash should default to 'hashtext(("group" || COALESCE(remote, ''''::text)))'
ok 116 - Column task.count should default to '0'
ok 117 - Column task.max should default to '(current_setting(''pg_task.default_max''::text, false))::integer'
ok 118 - Column task.state should default to 'PLAN'
ok 119 - Column task.delete should default to '(current_setting(''pg_task.default_delete''::text, false))::boolean'
ok 120 - Column task.drift should default to '(current_setting(''pg_task.default_drift''::text, false))::boolean'
ok 121 - Column task.header should default to '(current_setting(''pg_task.default_header''::text, false))::boolean'
ok 122 - Column task.string should default to '(current_setting(''pg_task.default_string''::text, false))::boolean'
ok 123 - Column task.delimiter should default to '(current_setting(''pg_task.default_delimiter''::text, false))::"char"'
ok 124 - Column task."group" should default to 'current_setting(''pg_task.default_group''::text, false)'
ok 125 - Column task."null" should default to 'current_setting(''pg_task.default_null''::text, false)'
ok 126 - Table task should not have a primary key
ok 127 - Column task(id) should not be a primary key
ok 128 - Column task(parent) should not be a primary key
ok 129 - Column task(plan) should not be a primary key
ok 130 - Column task(start) should not be a primary key
ok 131 - Column task(stop) should not be a primary key
ok 132 - Column task(live) should not be a primary key
ok 133 - Column task(timeout) should not be a primary key
ok 134 - Column task(repeat) should not be a primary key
ok 135 - Column task(hash) should not be a primary key
ok 136 - Column task(count) should not be a primary key
ok 137 - Column task(max) should not be a primary key
ok 138 - Column task(pid) should not be a primary key
ok 139 - Column task(state) should not be a primary key
ok 140 - Column task(delete) should not be a primary key
ok 141 - Column task(drift) should not be a primary key
ok 142 - Column task(header) should not be a primary key
ok 143 - Column task(string) should not be a primary key
ok 144 - Column task(delimiter) should not be a primary key
ok 145 - Column task(escape) should not be a primary key
ok 146 - Column task(quote) should not be a primary key
ok 147 - Column task(error) should not be a primary key
ok 148 - Column task("group") should not be a primary key
ok 149 - Column task(input) should not be a primary key
ok 150 - Column task("null") should not be a primary key
ok 151 - Column task(output) should not be a primary key
ok 152 - Column task(remote) should not be a primary key
ok 153 - Table public.task should be partitioned
